frontend/covid-dashboard-frontend/src/components/HomePage.js

```diff
@@ -37,155 +35,185 @@ const HomePage = ({cities, states, countries, loading, cityDetail, stateDetail,
   const [testingRateOptions, setTestingRateOptions] = useState({});
   const [isReady, setIsReady] = useState(false);
 
+  const chartContainerHeight = '375px';
 
+  const CITY_LOCATION = 'city';
+  const STATE_LOCATION = 'state';
+  const COUNTRY_LOCATION = 'country';
 
   useEffect(() => {
     refreshCharts();
-  }, []);
+  }, [cityDetail, stateDetail, countryDetail, loading]);
```

Comment:
added the detail arrays from the store into the dependency array of `useEffect` leads to the charts being refreshed anytime those update which would result once an action is dispatched during a location / location type change